src/d3a/d3a_core/redis_connections/redis_communication.py

```diff
@@ -69,16 +69,62 @@ def _subscribe_to_channels(self):
     def _reset_callback(self, _):
         self._simulation.reset()
 
-    def _stop_callback(self, _):
+    def _stop_callback(self, payload):
+        response = json.loads(payload["data"])
         self._simulation.stop()
+        stop_response_channel = f'{self._simulation_id}/response/stop'
+        if self._simulation.is_stopped:
```

Comment:
The publish of the response in all 3 commands could be refactored and reused. The publish_json calls are almost identical, and even the if / else blocks can be abstracted in a method that will be reused by all 3 endpoints.